Library/Homebrew/formula.rb

```diff
@@ -1836,10 +1835,10 @@ def self.full_names
   # this should only be used when users specify `--all` to a command
   # @private
   def self.all
-    # TODO: uncomment for 3.7.0 and ideally avoid using ARGV by moving to e.g. CLI::Parser
-    # if !ARGV.include?("--eval-all") && !Homebrew::EnvConfig.eval_all?
-    #   odeprecated "Formula#all without --all or HOMEBREW_EVAL_ALL"
-    # end
+    # TODO: ideally avoid using ARGV by moving to e.g. CLI::Parser
+    if ARGV.exclude?("--eval-all") && !Homebrew::EnvConfig.eval_all?
+      odeprecated "Formula#all without --all or HOMEBREW_EVAL_ALL"
+    end
```

Comment:
This is concerning this isn't addressed yet.

Particularly since as is this is going to break homebrew-command-not-found, the remove-disabled-formulae action and brew-pip-audit.

I feel a better option may be to just make this use the JSON API?

Reply:
@Bo98 Just a deprecation, not a disable.

> I feel a better option may be to just make this use the JSON API?

I agree but don't have the bandwidth to do that before the next major/minor release.

Reply:
> Particularly since as is this is going to break homebrew-command-not-found, the remove-disabled-formulae action and brew-pip-audit.

Also: they can move to effectively reimplementing this if desired/needed. We don't want to continue to provide the `Formula.all` API. Not sure how the JSON API fixes/changes that.